src/snowflake/snowpark/_internal/compiler/plan_compiler.py

```diff
@@ -77,98 +80,106 @@ def should_start_query_compilation(self) -> bool:
         )
 
     def compile(self) -> Dict[PlanQueryType, List[Query]]:
+        # initialize the queries with the original queries without optimization
+        final_plan = self._plan
+        queries = {
+            PlanQueryType.QUERIES: final_plan.queries,
+            PlanQueryType.POST_ACTIONS: final_plan.post_actions,
+        }
+
         if self.should_start_query_compilation():
-            session = self._plan.session
-            # preparation for compilation
-            # 1. make a copy of the original plan
-            start_time = time.time()
-            complexity_score_before_compilation = get_complexity_score(self._plan)
-            logical_plans: List[LogicalPlan] = [copy.deepcopy(self._plan)]
-            plot_plan_if_enabled(self._plan, "original_plan")
-            plot_plan_if_enabled(logical_plans[0], "deep_copied_plan")
-            deep_copy_end_time = time.time()
-
-            # 2. create a code generator with the original plan
-            query_generator = create_query_generator(self._plan)
-
-            extra_optimization_status: Dict[str, Any] = {}
-            # 3. apply each optimizations if needed
-            # CTE optimization
-            cte_start_time = time.time()
-            if session.cte_optimization_enabled:
-                repeated_subquery_eliminator = RepeatedSubqueryElimination(
-                    logical_plans, query_generator
+            try:
+                session = self._plan.session
```

Comment:
Did we have wrong indentation before introducing the `try` block?

Reply:
the indentation was correct. I would suggest using hide whitespace in diff-view settings. 

<img width="440" alt="Screenshot 2024-12-04 at 9 47 24 AM" src="https://github.com/user-attachments/assets/8fa5f6e9-95d9-4fdf-ae57-9c52c0f8e815">